File: 프로그래머스/1/82612. 부족한 금액 계산하기/부족한 금액 계산하기.cs
using System;

class Solution
{
    public long solution(int price, int money, int count)
    {
        long result = money;
        for(int i = 1;i < count +1 ; i++)
        {
            result -= price * i;
        }
        if(result > 0)
            result =0;
        else
            result *= -1;
        return result;
    }
}

File: 프로그래머스/1/82612. 부족한 금액 계산하기/부족한 금액 계산하기.java
import java.util.stream.IntStream;
class Solution {
    public long solution(int price, int money, int count) {
        long total = IntStream.rangeClosed(1, count)
            .mapToLong(n -> (long)n * price)
            .sum();
        
        return total >= money ? total - money : 0;
    }
}